Implementing basic FeedCategory model and tests

zach-delong opened the pull request:
This history (and diff) is much cleaner now.  I have a hunch of what happened to the last one, so I'll avoid that moving forward.

zach-delong committed:
Implementing basic FeedCategory model and tests in the new cleaner format.

jchristi commented on models/index.js:
```diff
@@ -6,12 +6,15 @@ var path = require('path');
   models.Feed.sync();
   models.Feed.belongsTo(models.User, { foreignKey: 'owner_uid' });
   models.AccessKey.belongsTo(models.User, { foreignKey: 'owner_uid' });
+  models.FeedCategory.belongsTo(models.User, { foreignKey: 'owner_uid' });
```
Missing association for ParentCategory (parent_cat)

jchristi commented:
Add association for parent category and a relevant test case(s)

zach-delong committed:
Associating FeedCategory with itself

- Added association in models/index.js to parent_cat
- Added test to ensure that model has association

ZacheryPD commented:
I went ahead and implemented the missing association, good catch!  Will this work, or do we also need an association to represent the other side of the relationship?  It looks like Sequelize can do both but I'm not familiar enough to say if we want to do that or not.

jchristi commented:
Whereas a category can only have one parent, a category can have many children. I think both associations should be defined.

zach-delong committed:
Adding Child associations to FeedCategory

- Added a has_many association to FeedCategory to represent a category's
children.
- Also renamed parent association to Parent to be more descriptive.

ZacheryPD commented:
@jchristi this should be resolved now.  Make sure you take a look at the names of the associations, did I do that right?  I think I did, but I expected that syntax to be different...

jchristi commented on tests/models/FeedCategory_test.js:
```diff
@@ -0,0 +1,68 @@
+      expect(sequelize.models.FeedCategory.associations).to.have.property('User');
+    }));
+
+    it('has a FedCategory (parent) property', co.wrap(function* (){
```
keep the test title simple: `has Parent property`

jchristi commented on tests/models/FeedCategory_test.js:
```diff
@@ -0,0 +1,68 @@
+      expect(sequelize.models.FeedCategory.associations).to.have.property('Parent');
+    }));
+
+    it('has a FedCategory (children) property', co.wrap(function* (){
```
`has Children property`

jchristi reviewed (changes requested):
minor text changes

jchristi commented:
@ZacheryPD the association names can be whatever we want. I'm debating whether they should be all lowercased or not.

Commit:
Updating test names based on feedback.
